_posts/news/2020-11-16-announcing-scalajs-1.3.1.md

```diff
@@ -0,0 +1,111 @@
+If you have been using the following setting to avoid bad performance on Firefox, you can now remove it:
+
+{% highlight scala %}
+// This was necessary before to avoid bad performance on Firefox
+scalaJSLinkerConfig ~= { _.withESFeatures(_.withUseECMAScript2015(false)) }
+{% endhighlight %}
+
+For the longer story: it turns out that SpiderMonkey, the JavaScript engine of Firefox, exhibits severe performance penalties for JavaScript `class`es.
+This has caused slowdowns for Scala.js code of up to 10x compared to other browsers and compared to the ECMAScript 5.1 mode.
+Starting from Scala.js 1.3.1, we restrict the usage of `class`es in the emitted JavaScript to the minimum required for correct semantics.
+We fall back to `function`s and `prototype`s for the majority of the emitted code.
+
+This change has a cost in terms of emitted code size.
+If your codebase does not target the Web nor any SpiderMonkey-based environment, you can tell Scala.js that avoiding `class`es is not necessary with the following setting:
+
+{% highlight scala %}
+// Trade away decent Firefox performance for a smaller output
+scalaJSLinkerConfig ~= { _.withESFeatures(_.withAvoidClasses(false)) }
+{% endhighlight %}
+
+## Miscellaneous
+
+### Better support of Scala 3 in sbt-scalajs
+
+The sbt plugin now has better support for Scala 3 codebases.
+In particular, the JUnit support works out of the box with `enablePlugins(ScalaJSJUnitPlugin)`.
+It is also future-proof for when sbt will support Scala 3 out of the box, rather than requiring `sbt-dotty`.
+
+These improvements are only available for Scala >= 3.0.0-M1.
+The 0.x versions are not supported.
+
+### New JDK APIs
+
+This release contains some improvements in the JDK APIs that we support:
+
+New classes:
+
+* `java.lang.ClassValue`
+
+New methods:
+
+* `new java.util.HashMap(initialCapacity: Int, loadFactor: Float)`
+* `java.util.Objects.requireNonNull(obj: T, message: Supplier[String])`
+
+Methods with fixed or improved behavior:
+
+* In `java.util.regex.Matcher`:
+  * `lookingAt()` and `matches()` ([issue #4252](https://github.com/scala-js/scala-js/issues/4252))
+  * `reset()` (issue [#4254](https://github.com/scala-js/scala-js/issues/4254))
+* `java.util.Formatter` and derivatives (such as `String.format`) now support `java.math.BigInteger` arguments with the `%d`, `%x` and `%o` conversions
+
+## Bug fixes
+
+Among others, the following bugs have been fixed in 1.3.1:
+
+* [#4252](https://github.com/scala-js/scala-js/issues/4252) regex.Matcher.lookingAt() and matches() incorrect in 1.3.0
+* [#4268](https://github.com/scala-js/scala-js/issues/4268) Native superclass of a non-native JS class is not imported (1.3.0 regression)
+* [#4271](https://github.com/scala-js/scala-js/issues/4271) Scala.js 1.3.0 fails to fastOptJS/fullOptJS when the project has no exports nor the module initializer
+* [#4254](https://github.com/scala-js/scala-js/issues/4254) Matcher.reset() does not reset the region
+
```

Comment:
Should scala-js/scala-js#4281 also be here? It could cause user visible changes.